libs/designer/src/lib/ui/templates/filters/templateFilters.tsx

```diff
@@ -4,16 +4,30 @@ import type { AppDispatch, RootState } from '../../../core/state/templates/store
 import { useIntl } from 'react-intl';
 import { useDispatch, useSelector } from 'react-redux';
 import { setConnectorsFilters, setDetailsFilters, setKeywordFilter, setSortKey } from '../../../core/state/templates/manifestSlice';
+import { useMemo } from 'react';
+import { getUniqueConnectorsFromConnections } from '../../../core/templates/utils/helper';
+import { useConnectorsOnly } from '../../../core/state/connection/connectionSelector';
 
 export interface TemplateFiltersProps {
-  connectors?: FilterObject[];
   detailFilters: Record<string, FilterObject[]>;
 }
 
-export const TemplateFilters = ({ connectors, detailFilters }: TemplateFiltersProps) => {
+export const TemplateFilters = ({ detailFilters }: TemplateFiltersProps) => {
   const dispatch = useDispatch<AppDispatch>();
   const sortKey = useSelector((state: RootState) => state?.manifest?.filters?.sortKey);
   const intl = useIntl();
+  const { availableTemplates, subscriptionId, location } = useSelector((state: RootState) => ({
+    availableTemplates: state.manifest.availableTemplates ?? {},
+    subscriptionId: state.workflow.subscriptionId,
+    location: state.workflow.location,
+  }));
+  const allTemplatesUniqueConnectorIds = useMemo(() => {
+    const allConnections = Object.values(availableTemplates).flatMap((template) => Object.values(template.connections));
+    const uniqueConnectorsFromConnections = getUniqueConnectorsFromConnections(allConnections, subscriptionId, location);
```

Comment:
You can just do .map here no, why do we need to new array again?



Just do `return getUniqueConnectorsFromConnections(allConnections, subscriptionId, location).map(connector => connector.connectorId)`

Reply:
I checked that the helper function that I reused (getUniqueConnectors) does not give back the unique connectors, so I used the Set to ensure the uniqueness of the ids

Reply:
I see that maybe this helper function was supposed to ensure the uniqueness. I think it's missing a logic; will fix it, that way I can remove this extra Set logic as well